number_theory/modular_unittest.cpp

```diff
@@ -1,41 +1,74 @@
+#include <stdint.h>
+
+#include <limits>
+#include <type_traits>
+
 #include <gtest/gtest.h>
 
 #include "number_theory/modular.h"
 
 namespace tql {
+namespace number_theory {
 
-TEST(ModularTest, Basic) {
-  using Mod10 = Modular<int, 10>;
+template <class T>
+void test_modular_basic() {
```

Comment:
It seems we are only testing assignments with explicit constructors. What about testing them implicitly too?